### ISSUE TITLE
[Question]: Chaining Locators 

### ISSUE THREAD

**Issue by infinitejest69** [microsoft/playwright#21217]
### Your question

TLDR:
can i chain ILocators

So this might be a me issue but i am refactoring my old component models and saw an opportunity to simplify my selectors by writing once selector to rather than repeating the same selector.

Example:
```html
<div class="complex-property">
    <mat-label class="ng-tns-c273-291">parent</mat-label>
    <button>
        <span class="mat-button-wrapper"><mat-icon svgicon="edit"><svg>
                    <path d="removed svg"></path>
                </svg></mat-icon></span></button>
</div>
```

the current way would be to write a locator for the edit button on the complex property 
```csharp
public ILocator complexpropertyEditButton(string name) => _page.Locator($".complex-property:has-text(\"{name}\") >> mat-icon[svgicon=\"edit\"] >> ../..");
```
And use it like
```csharp
await complexpropertyEditButton("parent").ClickAsync()
```

so the idea was to not have the button code all over the place for different components and have a generic component model for buttons and one for complex properties that can be chained together.
i am not sure if this is possible but i would like to chain my selectors so i can use the ILocator for edit button as a "sub locator" for items

```csharp
public ILocator ButtonEdit => _page.Locator("mat-icon[svgicon=\"edit\"] >> ../..");
public ILocator complexproperty => _page.Locator(".complex-property");
```

What i have tried:
This clicks the complex-property and not the edit button
```csharp
await complexproperty.Filter(new() { HasText = "parent"}).Filter(new() { Has = ButtonEdit }).ClickAsync();
```
if i try with the edit button first i cant filter up the node to make sure its the complex property that has text parent

Is it possible to not have to repeat the code? and simplify the selectors that allow chaining locators?
Something like this?
```csharp
await complexproperty.Locator(ButtonEdit ).ClickAsync();
```

**Comment by dgozman** [comment]
Fixed by #21391.

**Comment by infinitejest69** [comment]
Thanks. looking forward to using this change
